src/include/duckdb/execution/index/art/prefix.hpp

```diff
@@ -0,0 +1,53 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/execution/index/art/prefix.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include "duckdb/execution/index/art/art_key.hpp"
+#include "duckdb/storage/meta_block_writer.hpp"
+#include "duckdb/storage/meta_block_reader.hpp"
+
+namespace duckdb {
+class Prefix {
+public:
+	Prefix();
+	// Prefix created from key starting on `depth`.
+	Prefix(Key &key, uint32_t depth, uint32_t size);
+
+	// Returns the Prefix's size
+	uint32_t Size() const;
+
+	// Subscript operator
+	uint8_t &operator[](idx_t idx);
+
+	// Assign operator
+	Prefix &operator=(const Prefix &src);
+
+	// Move operator
+	Prefix &operator=(Prefix &&other) noexcept;
+
+	// Concatenate Prefix with a key and another prefix
+	// Used when deleting a Node.
+	// other.prefix + key + this->Prefix
+	void Concatenate(uint8_t key, Prefix &other);
+	// Reduces the prefix in n elements, and returns what would be the first one as a key
+	// I hate this function, looks messy
```

Comment:
Looks fine to me...?

Reply:
ah, this is an old misleading comment that I forgot to remove, of some initial version of it :-D